utils.go

```diff
@@ -65,12 +66,82 @@ func GetOffsetFromRequest(request *http.Request, options *Options) int64 {
 	return offset
 }
 
-// GenerateURI generates the pagination URI.
-func GenerateURI(limit int64, offset int64, options *Options) string {
+// GetCursorFromRequest returns current cursor.
+func GetCursorFromRequest(request *http.Request, options *Options) int64 {
+	var (
+		cursor int64
+		err    error
+	)
+
+	requestCursor := request.URL.Query().Get(options.CursorOptions.KeyName)
+
+	if requestCursor != "" {
+		cursor, err = strconv.ParseInt(requestCursor, 10, 64)
+		if err != nil {
+			cursor = 0
+		}
+	} else {
+		cursor = 0
+	}
+
+	return cursor
+}
+
+// GenerateOffsetURI generates the pagination URI.
+func GenerateOffsetURI(limit int64, offset int64, options *Options) string {
```

Comment:
As we don't handle `nil` with `options`, maybe better to pass `options Options` (so as a value) just for safer code.